src/StaticLibrary.h

```diff
@@ -22,6 +22,19 @@ namespace Internal {
 EXPORT void create_ar_file(const std::vector<std::string> &src_files, 
                            const std::string &dst_file, bool deterministic = true);
 
+/**
+ * Given a list of "files" (really, names and data), create an ar file.
+ * Note that 'data' is a string, but may hold binary data; do *not*
```

Comment:
The GPU kernels used to be stored in std::string, and something broke because of it, so we changed to `vector<char>` Probably due to c_str(), but maybe something else too.

I don't like it when comments are so important, I'd prefer `vector<char>` myself. Also, I'm surprised there is overhead, in my experience, this change involved negligible changes to the code, except having to explicitly pass a size/end iterator more often (which in the case of binary data, is more good than bad IMHO).